apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/app/observer/TrimObserver.java

```diff
@@ -52,16 +117,19 @@ boolean isTrimDerivedTables() {
     }
 
     private void processTrimEvent() {
+        log.trace("processTrimEvent() scheduled on previous run...");
         if (trimDerivedTables) {
             Integer trimHeight = null;
             synchronized (lock) {
                 if (trimDerivedTables) {
                     trimHeight = trimHeights.poll();
                 }
             }
-            if (trimHeight != null) {
-                log.debug("Perform trim on blockchain height {}", trimHeight);
+            if (trimHeight != null && trimHeight <= blockchain.getHeight()) {
```

Comment:
Peek first trimHeight value, check that proper height was reached and then remove from queue this value and perform trim

Reply:
Done